apm-agent-core/src/main/java/co/elastic/apm/agent/bci/classloading/ExternalPluginClassLoader.java

```diff
@@ -55,7 +58,27 @@ public ExternalPluginClassLoader(File pluginJar, ClassLoader agentClassLoader) t
             while (entries.hasMoreElements()) {
                 JarEntry jarEntry = entries.nextElement();
                 if (jarEntry.getName().endsWith(".class")) {
-                    tempClassNames.add(jarEntry.getName().replace('/', '.').substring(0, jarEntry.getName().length() - 6));
+                    String fqcn = jarEntry.getName().replace('/', '.').substring(0, jarEntry.getName().length() - 6);
+                    if (fqcn.startsWith("org.slf4j") || fqcn.startsWith("org.apache.logging")) {
```

Comment:
These are also in the list of agent dependencies so this case is not strictly needed. But I suppose you still opted for it because of the better error message, right?

Reply:
Right